plugins/woocommerce/src/Blocks/BlockTypes/ProductImage.php

```diff
@@ -219,19 +233,21 @@ protected function render( $attributes, $content, $block ) {
 		);
 
 		if ( $product ) {
+			$inner_content = $this->render_anchor(
+				$product,
+				$this->render_on_sale_badge( $product, $parsed_attributes ),
```

Comment:
While I can add the On Sale Badge in the Editor, it doesn't render on the frontend for some reason 🤔 (tried in Single Product block)

Reply:
Interesting, thanks for testing. I'll check it out. It renders on the Product Catalogue → Product Collection for me. Can you maybe verify that it renders there for you as well?